bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/votable/VoTableClient.scala

```diff
@@ -39,21 +42,31 @@ trait VoTableClient {
     p.future
   }
 
-  protected def doQuery(query: CatalogQuery, url: String): Future[QueryResult] = future {
-    val method = new GetMethod(s"$url/cgi-bin/conesearch.py")
-    val qs = queryParams(query)
-    method.setQueryString(qs)
+  // weak hash map should clean itself if we are using too much memory
+  val memo:Memo[(CatalogQuery, String), QueryResult] = Memo.weakHashMapMemo
```

Comment:
This is cool but I wonder about the `WeakHashMap` option in this case.  The idea with a `WeakHashMap` in general is to avoid the presence of the key in the map from keeping the key (and therefore the map entry) from being removed.  In this case there's nothing to keep the `(CatalogQuery, String)` pair from being removed at any time no?  That is nobody holds a reference to it.  I'm not sure whether the gc necessarily waits for memory pressure to clean that up.